### PR TITLE
implementation of Doubly Linked List in C

### PR DESCRIPTION
**Pull request by imsushant12**
… DIRECTORY

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

I have added the implementation of Doubly Linked List in C Language as well as updated README.md of the same.

### Checklist

- [YES ] I've read the contribution guidelines.
- [YES ] I've checked the issue list before deciding what to submit.
- [YES ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

https://github.com/TesseractCoding/NeoAlgo/issues/2323

### CODE REVIEW

File: C/ds/doubly_linked_list.c [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+        printf("\n----------------------------------");
+        printf("\n 1- To view the list.");
+        printf("\n 2- Insertion of element at starting.");
+        printf("\n 3- Insertion of element at end.");
+        printf("\n 4- Insertion of element at any position by providing previous element.");
+        printf("\n 5- Insertion of element at any position by providing index.");
+        printf("\n 6- Deletion of the first element.");
+        printf("\n 7- Deletion of the last element.");
+        printf("\n 8- Delete an element from any position.");
+        printf("\n 9- To exit.");
```
Comment (HarshCasper) [inline C/ds/doubly_linked_list.c:55]: Don't repeat them on every iteration. Instead keep them out of the loop body

File: C/ds/doubly_linked_list.c [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+                exit(1);
+                break;
+        default:
+            printf("\nIncorrect Choice!\n");
+        }
+    }
+    return 0;
+}
+
+void insertstart()
```
Comment (HarshCasper) [inline C/ds/doubly_linked_list.c:105]: Add a function description right here

File: C/ds/doubly_linked_list.c [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+    printf("\nEnter the element : ");
+    scanf("%d" , &data);
+    temp->info = data;
+    temp->prev = NULL;
+    temp->next = start;
+    start = temp;
+}
+
+
+void insertlast()
```
Comment (HarshCasper) [inline C/ds/doubly_linked_list.c:123]: Add a function description right here

File: C/ds/doubly_linked_list.c [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+    {
+        while(trav->next != NULL)
+            trav = trav->next;
+        temp->prev = trav;
+        trav->next = temp;
+    }
+}
+
+
+void insertafter(struct node *ptr)
```
Comment (HarshCasper) [inline C/ds/doubly_linked_list.c:154]: Add a function description right here

File: C/ds/doubly_linked_list.c [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+            if(temp->info == data)
+                return temp;
+            else
+                temp = temp->next;
+        }
+        return NULL;
+    }
+}
+
+void insertat()
```
Comment (HarshCasper) [inline C/ds/doubly_linked_list.c:196]: Add a function description right here

File: C/ds/doubly_linked_list.c [hunk 6]
```diff
@@ -0,0 +1,488 @@
...
+        newnode->next = temp->next;
+        newnode->prev = temp;
+        temp->next = newnode;
+        temp->next->prev = newnode;
+    }
+
+}
+
+
+void deletefirst()
```
Comment (HarshCasper) [inline C/ds/doubly_linked_list.c:232]: Add a function description right here

File: C/ds/doubly_linked_list.c [hunk 7]
```diff
@@ -0,0 +1,488 @@
...
+    {
+        temp = start;
+        start = start->next;
+        start->prev = NULL;
+        free(temp);
+    }
+}
+
+
+void deletelast()
```
Comment (HarshCasper) [inline C/ds/doubly_linked_list.c:251]: Add a function description right here

File: C/ds/doubly_linked_list.c [hunk 8]
```diff
@@ -0,0 +1,488 @@
...
+        else
+        {
+            temp->prev->next = NULL;
+            free(temp);
+        }
+    }
+}
+
+
+void deleteafter()
```
Comment (HarshCasper) [inline C/ds/doubly_linked_list.c:278]: Add a function description right here

File: C/ds/doubly_linked_list.c [hunk 9]
```diff
@@ -0,0 +1,488 @@
...
+        position = temp->next;
+        position->next->prev = temp;
+        temp->next = position->next;
+
+        free(position);
+    }
+}
+
+
+void viewlist()
```
Comment (HarshCasper) [inline C/ds/doubly_linked_list.c:308]: Add a function description right here

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make apt changes to the sample I/O as well. The Code looks good.

File: C/ds/doubly_linked_list.c (hunk 9 above)
Comment (imsushant12) [reply C/ds/doubly_linked_list.c:308]: Added!

File: C/ds/doubly_linked_list.c (hunk 8 above)
Comment (imsushant12) [reply C/ds/doubly_linked_list.c:278]: added

File: C/ds/doubly_linked_list.c (hunk 7 above)
Comment (imsushant12) [reply C/ds/doubly_linked_list.c:251]: added

File: C/ds/doubly_linked_list.c (hunk 4 above)
Comment (imsushant12) [reply C/ds/doubly_linked_list.c:154]: added

File: C/ds/doubly_linked_list.c (hunk 6 above)
Comment (imsushant12) [reply C/ds/doubly_linked_list.c:232]: added

File: C/ds/doubly_linked_list.c (hunk 2 above)
Comment (imsushant12) [reply C/ds/doubly_linked_list.c:105]: added

File: C/ds/doubly_linked_list.c (hunk 1 above)
Comment (imsushant12) [reply C/ds/doubly_linked_list.c:55]: did that

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few more changes: 

- Add a beginner friendly description about your Code
- Resolve Conflicts 

Otherwise good to merge :tada:

**Comment by imsushant12** [comment]
> Few more changes:
> 
> * Add a beginner friendly description about your Code
> * Resolve Conflicts
> 
> Otherwise good to merge 🎉

**Comment by imsushant12** [comment]
Okay working on it

**Review by HarshCasper — APPROVED** [review]

**Comment by HarshCasper** [comment]
@all-contributors bot please add @imsushant12 for code

**Pull request merged**